sdks/python/src/opik/evaluation/metrics/llm_judges/g_eval/metric.py

```diff
@@ -0,0 +1,82 @@
+from typing import Any, Optional, Union
+
+from opik.evaluation.metrics import base_metric, score_result
+from opik.evaluation.models import base_model, models_factory
+from .template import G_EVAL_COT_TEMPLATE, G_EVAL_QUERY_TEMPLATE
+from ... import exceptions
+
+
+class GEval(base_metric.BaseMetric):
+    def __init__(
+        self,
+        task_introduction: str,
+        evaluation_criteria: str,
+        model: Optional[Union[str, base_model.OpikBaseModel]] = None,
+        name: str = "g_eval_metric",
+    ):
+        super().__init__(
+            name=name,
+        )
+        self._init_model(model)
+
+        self.task_introduction = task_introduction
+        self.evaluation_criteria = evaluation_criteria
+
+        prompt = G_EVAL_COT_TEMPLATE.format(
+            task_introduction=self.task_introduction,
+            evaluation_criteria=self.evaluation_criteria,
+        )
+
+        self.llm_chain_of_thought: str = self._model.generate(input=prompt)
+
+    def _init_model(
+        self, model: Optional[Union[str, base_model.OpikBaseModel]]
+    ) -> None:
+        if isinstance(model, base_model.OpikBaseModel):
```

Comment:
As of today, OpikBaseModel supports only `str -> str` operations, and it's likely not going to be enough for this metric.

1. I think it's time to update `OpikBaseModel` to allow more flexibility in model usage and probably implement a new model class (I'd look at [litellm](https://github.com/BerriAI/litellm) since they provide a lot of models that you can use as if it was openai, later we can replace langchain model with litellm one in other metrics).
2. After such an update we should be able to use `logprobs`/`top_logprobs` and calculate an average score properly.

Without that, if we stick to current implementation, we'll have to make a lot of independent HTTP requests which will add a significant overhead, and the token usage is likely going to be significantly higher.